In [13]:
from nltk.translate.bleu_score import sentence_bleu
from nltk.translate.bleu_score import SmoothingFunction

def rouge_n_score(candidate, reference, n):
    smoothing = SmoothingFunction().method1
    candidate_tokens = candidate.split()
    reference_tokens = reference.split()
    return sentence_bleu([reference_tokens], candidate_tokens, smoothing_function=smoothing, weights=[1.0/n]*n)

def rouge_l_score(candidate, reference):
    smoothing = SmoothingFunction().method1
    candidate_tokens = candidate.split()
    reference_tokens = reference.split()
    return sentence_bleu([reference_tokens], candidate_tokens, smoothing_function=smoothing)

# Example usage
generated_summary = """nelong to the palace the forth evext that it Jinn which the building thy hindment a piece of the Wezeer to this She kissmoryThus was the But
the purse in heav and capting fears are period five by such But the eyes the noret
camelO ped the his wayBarain This este regates with the unnels in that if than time pisting The law is disses divinaned that which She answered with the kinds upll severalous
eyes but the
sage DZam chanted
omi graam languided anothen carries with respecting the same of inate bridegrols them had eated on the policrouled by EElJabndanlness in it the cageles of who have no the slave away the that the time of Palace rective for respeating till dend head
his fteent which I ridised every inder the sup is place pasrothed his powen deprinciption of the city called by God
felighted out madays fir bath to be of the Khaleefeh opening the piece me stuffs but God Hatten Hone shearing
fish childbe liberation enter performs serpent and rhe liad of the veil Jinn arm tishnus and had collected himself Whenefst he verilth the humpble twelve ordered at my hath never seen as Ibrheeem in their fram that he will that may fevautegaathenseo by the gazills it
and atnessed in the mids of a candled it calf the build garden nilter the monishmen  all what is knowlsed a branch is
after man should be in
amost ambnas cose that of the Prophet inother of the hands of salabueter fine that left on Book the Divination and came reason faste ElIsandfw Temred
the caomentres supply sifned
my caming he placesA ElBarder himself a prone whone of the
station her
hea large rion of Shereso O it if that they servant
 accovering the money and it said Jews I are dost there will be or that seen charable ods the ory having mob a be seas
prophecabid the soul
upon them of Birating a hade be only us know men when I had no the know  passhy take enter its in she came ammediately these nwered the intwo words which I baskerours haeay thou hast soncast tot the cappet happines and
exclad the must be night of a how hump and ets under what was the utled therefore thejson of methat Mousmy I as
to me and I wisten hath toreas highth he came utmost given to her
When the stee at the made forfiguished find he admissian assI have locks is to the fowl The mistress and is that of this man acquaintins of that happined for it is would
but God dit forrian rest the soo saidstrick the but he alth
lew it both I trousiers was is not cobuide the
close when the
King allso Sinder thee all pieted into a lamp manner or a melhin garby the coman and he into bitter its spacame heighty and the humpbacked his hampit was in he stly a honour may he be exalt for a
is shave too grind a punion
with it would part face hand ements opers that thou
hast enote night hath rewar She waterds om utterss she But more worly
of the Emes of gy the brought it is the  near the entered the
Nides they instruct it ell persired grast down and he wance and once Shaharriyr wails
osimplion the CONiutus Fusthy at this paid crass he Odation
th"""
reference_summary = """
To proceedThe lives of
former generations are a lesson
to posterity that a man may
review the remarkable events
which have happened to others and be admonished and may consider
the history of people of preceding ages and of all that hath
befallen them and be restrained Extolled be the perfection of
Him who hath thus ordained the history of former generations to be
a lesson to those which follow Such are the Tales of a Thousand
and One Nights with their romantic stories and their fables
It is related but God alone is allknowing as well as allwise
and almighty and allbountiful that there was in ancient times a
King of the countries of India and China possessing numerous troops
and guards and servants and domestic dependents and he had two
sons one of whom was a man of mature age and the other a youth
Both of these princes were brave horsemen but especially the elder
who inherited the kingdom of his father and governed his subjects
with such justice that the inhabitants of his country and whole empire
loved him He was called King Shahriyr his younger brother
was named ShhZemn and was King of Samarand The administration
of their governments was conducted with rectitude each
of them ruling over his subjects with justice during a period of twenty
years with the utmost enjoyment and happiness After this period
the elder King felt a strong desire to see his brother and ordered his
Wezeer to repair to him and bring him

Having taken the advice of the Wezeer on this subject he immediately
gave orders to prepare handsome presents such as horses
adorned with gold and costly jewels and memlooks and beautiful
virgins and expensive stuffs He then wrote a letter to his brother
expressive of his great desire to see him and having sealed it and
given it to the Wezeer together with the presents above mentioned
he ordered the minister to strain his nerves and tuck up his skirts
and use all expedition in returning The Wezeer answered without
delay I hear and obey and forthwith prepared for the journey he
packed his baggage removed the burdens and made ready all his
provisions within three days and on the fourth day he took leave
of the King Shahriyr and went forth towards the deserts and wastes
He proceeded night and day and each of the kings under the
authority of King Shahriyr by whose residence he passed came forth
to meet him with costly presents and gifts of gold and silver and
entertained him three days after which on the fourth day he
accompanied him one days journey and took leave of him Thus he
continued on his way until he drew near to the city of Samarand
when he sent forward a messenger to inform King ShhZemn of
his approach The messenger entered the city inquired the way to
the palace and introducing himself to the King kissed the ground
before him and acquainted him with the approach of his brothers
Wezeer upon which ShhZemn ordered the chief officers of his
court and the great men of his kingdom to go forth a days journey
to meet him and they did



"""
print(len(generated_summary))
print(len(reference_summary))


rouge1_score = rouge_n_score(generated_summary, reference_summary, 1)
rouge2_score = rouge_n_score(generated_summary, reference_summary, 2)
rougeL_score = rouge_l_score(generated_summary, reference_summary)

print("ROUGE-1 score:", rouge1_score)
print("ROUGE-2 score:", rouge2_score)
print("ROUGE-L score:", rougeL_score)


3000
3000
ROUGE-1 score: 0.32451499118165783
ROUGE-2 score: 0.11483468527414017
ROUGE-L score: 0.009538106189417589


In [ ]:
ROUGE scores tend to decrease as the length of the text increases, as it becomes more challenging for the generated summary to match the longer reference summary. In such cases, achieving high ROUGE scores, especially for ROUGE-2 and ROUGE-L, can be more challenging.

Considering that ROUGE-1 score is 0.33 and ROUGE-2 score is 0.12 for a 3000-character summary, it indicates some level of similarity between the generated summary and the reference summary. The ROUGE-L score of 0.01 indicates that the generated summary has some overlapping subsequences with the reference summary, though it may not capture the longest common subsequence effectively.

ROUGE scores tend to decrease as the length of the text increases, as it becomes more challenging for the generated summary to match the longer reference summary. In such cases, achieving high ROUGE scores, especially for ROUGE-2 and ROUGE-L, can be more challenging.

Considering that ROUGE-1 score is 0.33 and ROUGE-2 score is 0.12 for a 3000-character summary, it indicates some level of similarity between the generated summary and the reference summary. The ROUGE-L score of 0.01 indicates that the generated summary has some overlapping subsequences with the reference summary, though it may not capture the longest common subsequence effectively.

In [1]:
from nltk.translate.bleu_score import sentence_bleu, SmoothingFunction

def bleu_score(candidate, reference):
    smoothing = SmoothingFunction().method1
    candidate_tokens = candidate.split()
    reference_tokens = reference.split()
    return sentence_bleu([reference_tokens], candidate_tokens, smoothing_function=smoothing)


# Example usage
generated_summary = """nelong to the palace the forth evext that it Jinn which the building thy hindment a piece of the Wezeer to this She kissmoryThus was the But
the purse in heav and capting fears are period five by such But the eyes the noret
camelO ped the his wayBarain This este regates with the unnels in that if than time pisting The law is disses divinaned that which She answered with the kinds upll severalous
eyes but the
sage DZam chanted
omi graam languided anothen carries with respecting the same of inate bridegrols them had eated on the policrouled by EElJabndanlness in it the cageles of who have no the slave away the that the time of Palace rective for respeating till dend head
his fteent which I ridised every inder the sup is place pasrothed his powen deprinciption of the city called by God
felighted out madays fir bath to be of the Khaleefeh opening the piece me stuffs but God Hatten Hone shearing
fish childbe liberation enter performs serpent and rhe liad of the veil Jinn arm tishnus and had collected himself Whenefst he verilth the humpble twelve ordered at my hath never seen as Ibrheeem in their fram that he will that may fevautegaathenseo by the gazills it
and atnessed in the mids of a candled it calf the build garden nilter the monishmen  all what is knowlsed a branch is
after man should be in
amost ambnas cose that of the Prophet inother of the hands of salabueter fine that left on Book the Divination and came reason faste ElIsandfw Temred
the caomentres supply sifned
my caming he placesA ElBarder himself a prone whone of the
station her
hea large rion of Shereso O it if that they servant
 accovering the money and it said Jews I are dost there will be or that seen charable ods the ory having mob a be seas
prophecabid the soul
upon them of Birating a hade be only us know men when I had no the know  passhy take enter its in she came ammediately these nwered the intwo words which I baskerours haeay thou hast soncast tot the cappet happines and
exclad the must be night of a how hump and ets under what was the utled therefore thejson of methat Mousmy I as
to me and I wisten hath toreas highth he came utmost given to her
When the stee at the made forfiguished find he admissian assI have locks is to the fowl The mistress and is that of this man acquaintins of that happined for it is would
but God dit forrian rest the soo saidstrick the but he alth
lew it both I trousiers was is not cobuide the
close when the
King allso Sinder thee all pieted into a lamp manner or a melhin garby the coman and he into bitter its spacame heighty and the humpbacked his hampit was in he stly a honour may he be exalt for a
is shave too grind a punion
with it would part face hand ements opers that thou
hast enote night hath rewar She waterds om utterss she But more worly
of the Emes of gy the brought it is the  near the entered the
Nides they instruct it ell persired grast down and he wance and once Shaharriyr wails
osimplion the CONiutus Fusthy at this paid crass he Odation
th"""

reference_summary = """
To proceedThe lives of
former generations are a lesson
to posterity that a man may
review the remarkable events
which have happened to others and be admonished and may consider
the history of people of preceding ages and of all that hath
befallen them and be restrained Extolled be the perfection of
Him who hath thus ordained the history of former generations to be
a lesson to those which follow Such are the Tales of a Thousand
and One Nights with their romantic stories and their fables
It is related but God alone is allknowing as well as allwise
and almighty and allbountiful that there was in ancient times a
King of the countries of India and China possessing numerous troops
and guards and servants and domestic dependents and he had two
sons one of whom was a man of mature age and the other a youth
Both of these princes were brave horsemen but especially the elder
who inherited the kingdom of his father and governed his subjects
with such justice that the inhabitants of his country and whole empire
loved him He was called King Shahriyr his younger brother
was named ShhZemn and was King of Samarand The administration
of their governments was conducted with rectitude each
of them ruling over his subjects with justice during a period of twenty
years with the utmost enjoyment and happiness After this period
the elder King felt a strong desire to see his brother and ordered his
Wezeer to repair to him and bring him

Having taken the advice of the Wezeer on this subject he immediately
gave orders to prepare handsome presents such as horses
adorned with gold and costly jewels and memlooks and beautiful
virgins and expensive stuffs He then wrote a letter to his brother
expressive of his great desire to see him and having sealed it and
given it to the Wezeer together with the presents above mentioned
he ordered the minister to strain his nerves and tuck up his skirts
and use all expedition in returning The Wezeer answered without
delay I hear and obey and forthwith prepared for the journey he
packed his baggage removed the burdens and made ready all his
provisions within three days and on the fourth day he took leave
of the King Shahriyr and went forth towards the deserts and wastes
He proceeded night and day and each of the kings under the
authority of King Shahriyr by whose residence he passed came forth
to meet him with costly presents and gifts of gold and silver and
entertained him three days after which on the fourth day he
accompanied him one days journey and took leave of him Thus he
continued on his way until he drew near to the city of Samarand
when he sent forward a messenger to inform King ShhZemn of
his approach The messenger entered the city inquired the way to
the palace and introducing himself to the King kissed the ground
before him and acquainted him with the approach of his brothers
Wezeer upon which ShhZemn ordered the chief officers of his
court and the great men of his kingdom to go forth a days journey
to meet him and they did


"""
print(len(generated_summary))
print(len(reference_summary))

bleu_score = bleu_score(generated_summary, reference_summary)

print("BLEU score:", bleu_score)


3000
2999
BLEU score: 0.009538106189417589


In [2]:
import nltk
nltk.download('wordnet')


[nltk_data] Downloading package wordnet to /root/nltk_data...


True

In [8]:
import nltk
nltk.download('punkt')

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.


True

In [9]:
from nltk.translate import meteor_score
from nltk.tokenize import word_tokenize

generated_summary = """nelong to the palace the forth evext that it Jinn which the building thy hindment a piece of the Wezeer to this She kissmoryThus was the But
the purse in heav and capting fears are period five by such But the eyes the noret
camelO ped the his wayBarain This este regates with the unnels in that if than time pisting The law is disses divinaned that which She answered with the kinds upll severalous
eyes but the
sage DZam chanted
omi graam languided anothen carries with respecting the same of inate bridegrols them had eated on the policrouled by EElJabndanlness in it the cageles of who have no the slave away the that the time of Palace rective for respeating till dend head
his fteent which I ridised every inder the sup is place pasrothed his powen deprinciption of the city called by God
felighted out madays fir bath to be of the Khaleefeh opening the piece me stuffs but God Hatten Hone shearing
fish childbe liberation enter performs serpent and rhe liad of the veil Jinn arm tishnus and had collected himself Whenefst he verilth the humpble twelve ordered at my hath never seen as Ibrheeem in their fram that he will that may fevautegaathenseo by the gazills it
and atnessed in the mids of a candled it calf the build garden nilter the monishmen  all what is knowlsed a branch is
after man should be in
amost ambnas cose that of the Prophet inother of the hands of salabueter fine that left on Book the Divination and came reason faste ElIsandfw Temred
the caomentres supply sifned
my caming he placesA ElBarder himself a prone whone of the
station her
hea large rion of Shereso O it if that they servant
 accovering the money and it said Jews I are dost there will be or that seen charable ods the ory having mob a be seas
prophecabid the soul
upon them of Birating a hade be only us know men when I had no the know  passhy take enter its in she came ammediately these nwered the intwo words which I baskerours haeay thou hast soncast tot the cappet happines and
exclad the must be night of a how hump and ets under what was the utled therefore thejson of methat Mousmy I as
to me and I wisten hath toreas highth he came utmost given to her
When the stee at the made forfiguished find he admissian assI have locks is to the fowl The mistress and is that of this man acquaintins of that happined for it is would
but God dit forrian rest the soo saidstrick the but he alth
lew it both I trousiers was is not cobuide the
close when the
King allso Sinder thee all pieted into a lamp manner or a melhin garby the coman and he into bitter its spacame heighty and the humpbacked his hampit was in he stly a honour may he be exalt for a
is shave too grind a punion
with it would part face hand ements opers that thou
hast enote night hath rewar She waterds om utterss she But more worly
of the Emes of gy the brought it is the  near the entered the
Nides they instruct it ell persired grast down and he wance and once Shaharriyr wails
osimplion the CONiutus Fusthy at this paid crass he Odation
th"""

reference_summary = """
To proceedThe lives of
former generations are a lesson
to posterity that a man may
review the remarkable events
which have happened to others and be admonished and may consider
the history of people of preceding ages and of all that hath
befallen them and be restrained Extolled be the perfection of
Him who hath thus ordained the history of former generations to be
a lesson to those which follow Such are the Tales of a Thousand
and One Nights with their romantic stories and their fables
It is related but God alone is allknowing as well as allwise
and almighty and allbountiful that there was in ancient times a
King of the countries of India and China possessing numerous troops
and guards and servants and domestic dependents and he had two
sons one of whom was a man of mature age and the other a youth
Both of these princes were brave horsemen but especially the elder
who inherited the kingdom of his father and governed his subjects
with such justice that the inhabitants of his country and whole empire
loved him He was called King Shahriyr his younger brother
was named ShhZemn and was King of Samarand The administration
of their governments was conducted with rectitude each
of them ruling over his subjects with justice during a period of twenty
years with the utmost enjoyment and happiness After this period
the elder King felt a strong desire to see his brother and ordered his
Wezeer to repair to him and bring him

Having taken the advice of the Wezeer on this subject he immediately
gave orders to prepare handsome presents such as horses
adorned with gold and costly jewels and memlooks and beautiful
virgins and expensive stuffs He then wrote a letter to his brother
expressive of his great desire to see him and having sealed it and
given it to the Wezeer together with the presents above mentioned
he ordered the minister to strain his nerves and tuck up his skirts
and use all expedition in returning The Wezeer answered without
delay I hear and obey and forthwith prepared for the journey he
packed his baggage removed the burdens and made ready all his
provisions within three days and on the fourth day he took leave
of the King Shahriyr and went forth towards the deserts and wastes
He proceeded night and day and each of the kings under the
authority of King Shahriyr by whose residence he passed came forth
to meet him with costly presents and gifts of gold and silver and
entertained him three days after which on the fourth day he
accompanied him one days journey and took leave of him Thus he
continued on his way until he drew near to the city of Samarand
when he sent forward a messenger to inform King ShhZemn of
his approach The messenger entered the city inquired the way to
the palace and introducing himself to the King kissed the ground
before him and acquainted him with the approach of his brothers
Wezeer upon which ShhZemn ordered the chief officers of his
court and the great men of his kingdom to go forth a days journey
to meet him and they did
"""

# Tokenize the sentences
generated_summary_tokens = word_tokenize(generated_summary)
reference_summary_tokens = word_tokenize(reference_summary)

# Calculate the METEOR score
meteor_score = meteor_score.meteor_score([reference_summary_tokens], generated_summary_tokens)

# Print the METEOR score
print("METEOR score:", meteor_score)


METEOR score: 0.20092592592592592


A METEOR score of 0.2009 means that the generated summary has a moderate level of similarity and alignment with the reference summary.
